plugin/pkg/admission/priority/admission_test.go

```diff
@@ -189,6 +189,14 @@ func TestDefaultPriority(t *testing.T) {
 			expectedDefaultBefore: scheduling.DefaultPriorityWhenNoDefaultClassExists,
 			expectedDefaultAfter:  defaultClass1.Value,
 		},
+		{
+			name:                  "multiple default classes resolves to the minimum value among them",
+			classesBefore:         []*scheduling.PriorityClass{defaultClass1, defaultClass2},
+			classesAfter:          []*scheduling.PriorityClass{defaultClass2},
+			attributes:            admission.NewAttributesRecord(nil, nil, pcKind, "", defaultClass1.Name, pcResource, "", admission.Delete, nil),
+			expectedDefaultBefore: defaultClass1.Value,
+			expectedDefaultAfter:  defaultClass2.Value,
+		},
```

Comment:
Can we have one more test case with 2 default priorityClasses with same priority value then the first one is choosen(as per the current logic).

Reply:
When the priority values are the same, how do you want to verify that the first one is picked? Admission controller does not update the "priorityClassName" field. It only sets the value of priority.

Reply:
Apologies, I wrote my earlier comment in haste. I was under the impression that this tests getDefaultPriorityClass(), it seems we are checking only getDefaultPriority() f(n) in TestDefaultPriority, so we will return always the same value but if we can call getDefaultPriorityClass() f(n) in this test, we can check for the priorityClass being used as well.

Reply:
I see your point, but getDefaultPriorityClass is used for only two purposes: 1. Whether any default priority classes exists or not, 2. Find the value of default priority. That's why it does not matter which priority class it returns when there are two default priority classes with the same value.
Returning the value of the first priority class is a side effect of the current implementation. It is not necessary for the validity of the logic. For this reason, I believe we should not add a test for it. Having a test may make users and developers think that returning the first priority is a necessary part of the logic. Besides, I am not sure if a `lister` is guaranteed to return the list of items always in the same order.

Reply:
Thanks for the clarification @bsalamat. Yeah, I think that it is a side-effect because we are using a list where order matters.